src/main/scala/kinesis/mock/api/PutRecordsRequest.scala

```diff
@@ -120,6 +117,18 @@ final case class PutRecordsRequest(
 }
 
 object PutRecordsRequest {
+  private def getIndexByShard(
```

Comment:
One of the purposes of the groupBy was to assign an index number for each record that would be placed on a shard via zipWithIndex. This basically does that but without the groupBy, preserving the order of the input records